pkg/servicemonitor/servicemonitor.go

```diff
@@ -53,52 +70,67 @@ func (u *ServiceMonitor) UpdateServiceMonitorDeployment(template monitoringv1.Se
-	if serviceMonitorRef == (v1alpha1.NamespacedName{}) {
-		return nil
-	}
-	namespacedName := types.NamespacedName{Name: serviceMonitorRef.Name, Namespace: serviceMonitorRef.Namespace}
-	resource := &monitoringv1.ServiceMonitor{}
-	// Does the resource already exist?
-	err := u.Client.Get(u.Ctx, namespacedName, resource)
+// Creates or Updates Service Monitor Deployment according to the template if enable of the hypershift
+func (u *ServiceMonitor) HypershiftUpdateServiceMonitorDeployment(template rhobsv1.ServiceMonitor) error {
+	namespacedName := types.NamespacedName{Name: template.Name, Namespace: template.Namespace}
+	deployedServiceMonitor := &rhobsv1.ServiceMonitor{}
+	err := u.Client.Get(u.Ctx, namespacedName, deployedServiceMonitor)
 	if err != nil {
+		// No similar ServiceMonitor exists
 		if !k8serrors.IsNotFound(err) {
-			// If this is an unknown error
 			return err
 		}
-		// Resource doesn't exist, nothing to do
-		return nil
+		return u.Client.Create(u.Ctx, &template)
 	}
-	return u.Client.Delete(u.Ctx, resource)
-}
-
-func (u *ServiceMonitor) GetServiceMonitor(namespacedName types.NamespacedName) (monitoringv1.ServiceMonitor, error) {
-	fmt.Println("GetClusterID")
-	serviceMonitor := monitoringv1.ServiceMonitor{}
-	err := u.Client.Get(u.Ctx, namespacedName, &serviceMonitor)
-	return serviceMonitor, err
+	if !u.Comparer.DeepEqual(deployedServiceMonitor.Spec, template.Spec) {
+		// Update existing ServiceMonitor for the case that the template changed
+		deployedServiceMonitor.Spec = template.Spec
+		return u.Client.Update(u.Ctx, deployedServiceMonitor)
+	}
+	return nil
 }
 
-// TemplateForServiceMonitorResource returns a ServiceMonitor
-func TemplateForServiceMonitorResource(url, blackBoxExporterNamespace string, namespacedName types.NamespacedName, clusterID string) monitoringv1.ServiceMonitor {
-
-	routeURL := url
-
-	routeMonitorLabels := blackboxexporter.GenerateBlackBoxExporterLables()
+// Deletes the ServiceMonitor Deployment
+func (u *ServiceMonitor) DeleteServiceMonitorDeployment(serviceMonitorRef v1alpha1.NamespacedName) error {
+	if serviceMonitorRef == (v1alpha1.NamespacedName{}) {
+		return nil
+	}
+	namespacedName := types.NamespacedName{Name: serviceMonitorRef.Name, Namespace: serviceMonitorRef.Namespace}
 
-	labelSelector := metav1.LabelSelector{MatchLabels: routeMonitorLabels}
+	if u.Hypershift == true {
+		resource := &rhobsv1.ServiceMonitor{}
+		// Does the resource already exist?
+		err := u.Client.Get(u.Ctx, namespacedName, resource)
+		if err != nil {
+			if !k8serrors.IsNotFound(err) {
+				// If this is an unknown error
+				return err
+			}
+			// Resource doesn't exist, nothing to do
+			return nil
+		}
 
-	// Currently we only support `http_2xx` as module
-	// Still make it a variable so we can easily add functionality later
-	modules := []string{"http_2xx"}
+		return u.Client.Delete(u.Ctx, resource)
+	} else {
```

Comment:
This else can be removed liked the above example since there is a return contained in the if statement above